subprojects/execution/src/main/java/org/gradle/internal/execution/steps/AbstractValidateStep.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.execution.steps;
+
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableSet;
+import org.gradle.internal.execution.UnitOfWork;
+import org.gradle.internal.execution.WorkValidationContext;
+import org.gradle.internal.execution.WorkValidationException;
+import org.gradle.internal.operations.BuildOperationExecutor;
+import org.gradle.internal.reflect.validation.TypeValidationProblem;
+import org.gradle.internal.reflect.validation.TypeValidationProblemRenderer;
+import org.gradle.model.internal.type.ModelType;
+
+import java.util.Collection;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Abstract class for steps that validate a {@link UnitOfWork} or report validation errors present in the {@link WorkValidationContext}.
+ *
+ * @param <C> previous execution context
+ * @param <R> result of execution delegate
+ */
+/* package */ abstract class AbstractValidateStep<C extends Context, R extends Result> extends BuildOperationStep<C, R> {
```

Comment:
Is this `/* package */` notation something we do now?

Reply:
I didn't even think to check if that was a Gradle standard, I've just been doing it for awhile.  I think it's especially helpful when switching between Groovy and Java contexts as an affirmation that this class is meant to use the default access level.  It makes readers very aware that this isn't a class that was meant to be `public` where the author forgot the need to be explicit which  currently just happens to work with either access level.

That said, I'll drop it if you think it's too much of a clash with our current style.